Source/vtkPyFR/vtkPyFRPipeline.cxx

```diff
@@ -688,31 +782,35 @@ int vtkPyFRPipeline::CoProcess(vtkCPDataDescription* dataDescription)
       vtkWeakPointer<vtkRenderWindow> rw = viewProxy->GetRenderWindow();
       vtkWeakPointer<vtkRendererCollection> rens = rw->GetRenderers();
       assert(rens);
-      vtkWeakPointer<vtkRenderer> ren = rens->GetFirstRenderer();
-      ren->ResetCamera();
-      ren->ResetCameraClippingRange();
-      vtkWeakPointer<vtkCamera> cam = ren->GetActiveCamera();
-      const float* eye = pyd->eye;
-      if(!std::isnan(eye[0]) && !std::isnan(eye[1]) && !std::isnan(eye[2])) {
-        cam->SetPosition(eye[0], eye[1], eye[2]);
-      }
-      const float* ref = pyd->ref;
-      if(!std::isnan(ref[0]) && !std::isnan(ref[1]) && !std::isnan(ref[2])) {
-        cam->SetFocalPoint(ref[0], ref[1], ref[2]);
-      }
-      const float* vup = pyd->vup;
-      if(!std::isnan(vup[0]) && !std::isnan(vup[1]) && !std::isnan(vup[2])) {
-        cam->SetViewUp(vup[0], vup[1], vup[2]);
-      }
-      const float* bg = pyd->bg_color;
-      if(!std::isnan(bg[0]) && !std::isnan(bg[1]) && !std::isnan(!bg[2])) {
-        ren->SetBackground(bg[0], bg[1], bg[2]);
+      if(!i)
+      for(int j=0;j<rens->GetNumberOfItems();j++)
```

Comment:
The loop is needed to make sure that all views have the same background. The loop could be optimized by doing some calculations once outside.